### PR TITLE
Add async support to fetch pages

### PR DESCRIPTION
**Pull request by dbaxa**
Sem-Ver: feature

### CODE REVIEW

File: jira/resilientsession.py [hunk 1]
```diff
@@ -57,12 +57,6 @@ def raise_on_error(r, verb='???', **kwargs):
     # for debugging weird errors on CI
     if r.status_code not in [200, 201, 202, 204]:
         raise JIRAError(r.status_code, request=request, response=r, **kwargs)
-    # testing for the WTH bug exposed on
```
Comment (ssbarnea) [inline jira/resilientsession.py]: Why are you removing this? This is a workaround for https://jira.atlassian.com/browse/JRASERVER-41559 which was never fixed.

Comment (dbaxa) [reply jira/resilientsession.py]: @ssbarnea perhaps I misunderstood the code that I removed. It looked to me like it wasn't actually doing anything.

Comment (ssbarnea) [reply jira/resilientsession.py]: Remove any change that is unrelated to this feature and I will merge it as it seems to have its own tests which are passing.

Comment (dbaxa) [reply jira/resilientsession.py]: @ssbarnea will do. I'll break the other changes out into individual prs.

File: jira/utils/__init__.py [hunk 2]
```diff
@@ -77,8 +76,10 @@ def threaded_requests(requests):
 
 def json_loads(r):
     raise_on_error(r)
-    if len(r.text):  # r.status_code != 204:
-        return json.loads(r.text)
-    else:
+    try:
+        return r.json()
```
Comment (dbaxa) [inline jira/utils/__init__.py]: @ssbarnea I'd very much like to keep this change / make this change as well. If it needs to go in another PR please let me know :-).

**Comment by dbaxa** [comment]
@ssbarnea can we merge this ?
